altair/api.py

```diff
@@ -282,6 +284,32 @@ def encode(self, **kwargs):
         self.encoding = Encoding(**kwargs)
         return self
 
+    def configure(self, **kwargs):
+        """Set chart configuration"""
+        self.vlconfig = VLConfig(**kwargs)
+        return self
+
+    def set_single_dims(self, width, height):
```

Comment:
I worried a bit about making this behavior automatic for a couple reason:

* The relationship between width/height and singleWidth/singleHeight will then be largely implicit
* This only really works well about half the time (see all the manual fiddling in the Notebook in this PR). 

At this point, users have to understand the (complex) relationship between width/height, singleWidth/singleHeight, and `band.size` to get their charts rendering correctly. 

I think we *can* abstract this away, but its going to take a lot of work. For example, an aggregation can force the number of ordinal/nominal bands that get created, and then the user will need to fiddle with `band.size` to get the exact dimensioning they need.